collections/src/commonMain/kotlin/SynchronizedMap.kt

```diff
@@ -0,0 +1,125 @@
+package com.juul.tuulbox.collections
+
+import kotlinx.atomicfu.locks.ReentrantLock
+import kotlinx.atomicfu.locks.reentrantLock
+import kotlinx.atomicfu.locks.withLock
+
+/** A [MutableMap] where all reads and writes are protected by a [ReentrantLock]. */
+public class SynchronizedMap<K, V>(
+    private val inner: MutableMap<K, V>,
```

Comment:
Would it make sense to provide a default for this? Or is that too opinionated?

```suggestion
    private val inner: MutableMap<K, V> = mutableMapOf(),
```

Reply:
Did one better, and made public constructors analogous to those provided for standard library maps, including `synchronizedMapOf` functions.